spec/index.bs

```diff
@@ -3693,21 +3693,111 @@ GPURenderPipeline includes GPUPipelineBase;
+                - assign the |vertexList| item |i| to be [$fetch index$](|i| + |drawCall|.firstIndex, |indexBuffer|.buffer, |indexBuffer|.offset, |indexBuffer|.format) + |drawCall|.baseVertex
+            If |vertexState|.{{GPUVertexStateDescriptor/indexFormat}} is `null`:
+                - initialize the |vertexList| with |drawCall|.vertexCount integers
+                - assign the |vertexList| item |i| to be |drawCall|.firstVertex + |i|
+
+        1. Process vertices. Each vertex |vertexIndex| in the |vertexList|,
+            in each instance of index |rawInstanceIndex| in range from 0 to |drawCall|.instanceCount - 1, inclusive,
+            is processed separately:
+            1. Let |instanceIndex| be |rawInstanceIndex| + |drawCall|.baseInstance.
+            1. For each non-`null` |vertexBufferDesc| under index |i|
+                in the list of |vertexState|.{{GPUVertexStateDescriptor/vertexBuffers}}:
+                1. Let |vertexBuffer| and |vertexBufferOffset| be the buffer and offset in |vertexBuffers| bindings at slot |i|
+                1. Let |vertexElementIndex| be dependent on |vertexBufferDesc|.{{GPUVertexBufferLayoutDescriptor/stepMode}}:
+                    <dl class="switch">
+                        : {{GPUInputStepMode/"vertex"}}
+                        :: |vertexIndex|
+                        : {{GPUInputStepMode/"instance"}}
+                        :: |instanceIndex|
+                    </dl>
+                1. For each |attributeDesc| in |vertexBufferDesc|.{{GPUVertexBufferLayoutDescriptor/attributes}}:
+                    1. Let |attributeOffset| be |vertexBufferOffset| +
+                        |vertexElementIndex| * |vertexBufferDesc|.{{GPUVertexBufferLayoutDescriptor/arrayStride}} +
+                        |attributeDesc|.{{GPUVertexAttributeDescriptor/offset}}.
+                    1. Load the attribute |data| of format |attributeDesc|.{{GPUVertexAttributeDescriptor/format}}
+                        from |vertexBuffer| starting at offset |attributeOffset|.
+                    1. Convert the |data| into a shader-visible format.
+                        <div class="example">
+                        An attribute of type {{GPUVertexFormat/"uchar2norm"}} will be converted from
+                            2 bytes of fixed-point unsigned 8-bit integers into 2 floating-point values
+                            as `vec2<f32>` in WGSL.
+                        </div>
+                    1. Bind the |data| to vertex shader input
+                        location |attributeDesc|.{{GPUVertexAttributeDescriptor/shaderLocation}}.
+            1. For each {{GPUBindGroup}} group at |index| in |bindGroups|:
+                1. Bind the group of resources visible to vertex shader group |index|.
+            1. Invoke vertex shader in |descriptor|.{{GPURenderPipelineDescriptor/vertexStage}}.
+
+                Note: The target platform caches the results of vertex shader invocations.
```

Comment:
And similarly a vertexIndex may be processed more times than it is listed, for example when breaking up long strips.